motion_computation/src/psm_to_external_object_convertor.cpp

```diff
@@ -306,16 +318,14 @@ std::vector<geometry_msgs::msg::Pose> sample_2d_linear_motion(
   double total_dt = 0;
 
   while (total_dt < period) {
-    // Compute the 2d position and orientation in the Pose frame
+    // Increment time
     total_dt += step_size;
-    double dx_from_start = velocity * total_dt;  // Assuming linear motion in pose frame
+    double distance = velocity * total_dt;  // Total distance traveled
 
-    double x = pose.position.x + dx_from_start;
```

Comment:
Still trying to wrap my head around this but putting down my though - the pose.position.x in the map frame should be a fairly large number - having an error of this size should be noticeable.

Reply:
Correct. It was wrong because it is arbitrarily adding the x value of the original pose, which is already captured by the first transform.